source/_integrations/mqtt.markdown

```diff
@@ -559,44 +559,233 @@ The following software has built-in support for MQTT discovery:
+
+To set up a any device with multiple entities, a Configuration topic must be published for each entity. Each Configuration topic must include a "unique_id" property which is unique to the entity being created, and a "device" property which includes a device "identifiers" property which is unique to the device.
+
+A user, script, or device would send its configuration as JSON payload to the Configuration topic. After the first message to `config`, then the device may send MQTT messages with new sensor measurements in the form of a JSON payload.
+
+Set up a device with two sensor entities:
+
+- Configuration topic 1: `homeassistant/sensor/my_custom_sensor_001_temperature/config`
+- Configuration payload 1:
+```json
+{
+    "name": "My Custom Sensor 001 Temperature",
+    "unique_id": "my_custom_sensor_001_temperature",
+    "state_topic": "my_state_topic/my_custom_sensor/001",
+    "value_template": {% raw %} "{{ value_json.temperature }}" {% endraw %},
+    "device": {
+        "identifiers": ["my_custom_sensor_001"],
+        "name": "My Custom Sensor 001",
+        "model": "My Model Number",
+        "manufacturer": "My Manufacturer ID"
+    }
+}
+```
+- Configuration topic 2: `homeassistant/sensor/my_custom_sensor_001_humidity/config`
+- Configuration payload 2: 
+```json
+{
+    "name": "My Custom Sensor 001 Humidity",
+    "unique_id": "my_custom_sensor_001_humidity",
+    "state_topic": "my_state_topic/my_custom_sensor/001",
+    "value_template": {% raw %} "{{ value_json.humidity }}" {% endraw %},
+    "device": {
+        "identifiers": ["my_custom_sensor_001"],
+        "name": "My Custom Sensor 001",
+        "model": "My Model Number",
+        "manufacturer": "My Manufacturer ID"
+    }
+}
+```
+- State topic: `my_state_topic/my_custom_sensor/001`
+- State payload: `{ "temperature": 23.20, "humidity": 43.70 }`
+
+Notes:
+- The configuration topic must be unique for each entity. In this example the configuration topic 'object_id' combines the device name, serial number, and sensor id to form a unique topic.
+- Don't forget to publish configuration messages with 'retain' set to true.
+- If publishing the configuration payload via the Home Assistant MQTT integration (Settings > Devices and Services > core-mosquitto > Configure), be sure to tick 'Allow template'.
+
+Delete this example device and entities:
+- Configuration topic 1: `homeassistant/sensor/my_custom_sensor_001_temperature/config`
+- Configuration payload 1: (empty)
+- Configuration topic 2: `homeassistant/sensor/my_custom_sensor_001_humidity/config`
+- Configuration payload 2: (empty)
+- The messages must be published 'retain' set to true. Do not tick 'Allow template'.
+- If using mosquitto_pub, see previous example.
+
+#### Using a script to create a device with multiple sensor entities (door/window sensors)
+
```

Comment:
Let us not add this here

Reply:
This is some to publish on the Home Assistant community, but I assume you already did that.